pkg/recommendation/factory.go

```diff
@@ -34,36 +34,46 @@ func NewFactory(logger logrus.FieldLogger, dynamicCli dynamic.Interface) *Factor
 }
 
 // NewForSources merges recommendation options from multiple sources, and creates a new Set.
-func (f *Factory) NewForSources(sources map[string]config.Sources) Set {
+func (f *Factory) NewForSources(sources map[string]config.Sources, mapKeyOrder []string) Set {
 	set := make(map[string]Recommendation)
-	for _, source := range sources {
+	for _, key := range mapKeyOrder {
+		source, exists := sources[key]
+		if !exists {
+			continue
+		}
+
 		k8sSource := source.Kubernetes
-		f.createIfNotExist(k8sSource.Recommendations.Pod.LabelsSet, set,
+		f.enableRecommendationIfShould(k8sSource.Recommendations.Pod.LabelsSet, set,
 			func() Recommendation { return NewPodLabelsSet() },
 		)
-		f.createIfNotExist(k8sSource.Recommendations.Pod.NoLatestImageTag, set,
+		f.enableRecommendationIfShould(k8sSource.Recommendations.Pod.NoLatestImageTag, set,
 			func() Recommendation { return NewPodNoLatestImageTag() },
 		)
-		f.createIfNotExist(k8sSource.Recommendations.Ingress.BackendServiceValid, set,
+		f.enableRecommendationIfShould(k8sSource.Recommendations.Ingress.BackendServiceValid, set,
 			func() Recommendation { return NewIngressBackendServiceValid(f.dynamicCli) },
 		)
-		f.createIfNotExist(k8sSource.Recommendations.Ingress.TLSSecretValid, set,
+		f.enableRecommendationIfShould(k8sSource.Recommendations.Ingress.TLSSecretValid, set,
 			func() Recommendation { return NewIngressTLSSecretValid(f.dynamicCli) },
 		)
 	}
 	return newRecommendationsSet(f.logger, set)
```

Comment:
do we need to really have a set? what about `[]Recommendation` so that would be the `enableRecommendationIfShould` output:
```go
func (f *Factory) enableRecommendationIfShould(rec config.Recommendations) []Recommendation {
	var list []Recommendation
	if isTrue(rec.Pod.LabelsSet) {
		list = append(list, NewPodLabelsSet())
	}

	if isTrue(rec.Pod.NoLatestImageTag) {
		list = append(list, NewPodNoLatestImageTag())

	}

	if isTrue(rec.Ingress.BackendServiceValid) {
		list = append(list, NewIngressBackendServiceValid(f.dynamicCli))

	}

	if isTrue(rec.Ingress.TLSSecretValid) {
		list = append(list, NewIngressTLSSecretValid(f.dynamicCli))
	}

	return list
}
```

Reply:
Sure, if we already ensured there are single instances of each validator, then yes, it can be a slice 👍